lib/prelude.dx

```diff
@@ -537,6 +545,30 @@ def sq  [Mul a] (x:a) : a = x * x
 def abs [Add a, Ord a] (x:a) : a = select (x > zero) x (zero - x)
 def mod (x:Int) (y:Int) : Int = rem (y + rem x y) y
 
+instance [Floating a] Floating (n=>a)
+  exp    = map exp
+  exp2   = map exp2
+  log    = map log
+  log2   = map log2
+  log10  = map log10
+  log1p  = map log1p
+  sin    = map sin
+  cos    = map cos
+  tan    = map tan
+  sinh   = map sinh
+  cosh   = map cosh
+  tanh   = map tanh
+  floor  = map floor
+  ceil   = map ceil
+  round  = map round
+  sqrt   = map sqrt
+  pow    = \x y. for i. pow x.i y.i
+  lgamma = map lgamma
+
+def axis1 (x : a => b => c) : b => a => c = for j. for i. x.i.j
```

Comment:
It might be a good idea to use `view` instead of `for`, for more predictable performance. With `for`, there's a risk that we'll materialize the permuted table, which might not be what you want.

Reply:
Oh I didn't even know about `view`. That's good to know.

Reply:
Not surprising. We added it [less than two weeks ago!](https://github.com/google-research/dex-lang/commit/5be588fc14d981b57028b550934f9c7b1523498b)